qcodes/tests/dataset/test_dataset_in_memory.py

```diff
@@ -43,6 +46,124 @@ def test_dataset_in_memory_reload_from_db(
     compare_datasets(ds, loaded_ds)
 
 
+@settings(
+    deadline=None,
+    suppress_health_check=(HealthCheck.function_scoped_fixture,),
+    max_examples=10,
+)
+@given(
+    shape1=hst.integers(min_value=1, max_value=100),
+    shape2=hst.integers(min_value=1, max_value=100),
+)
+def test_dataset_in_memory_reload_from_db_2d(
+    meas_with_registered_param_2d, DMM, DAC, tmp_path, shape1, shape2
+):
+    meas_with_registered_param_2d.set_shapes(
+        {
+            DMM.v1.full_name: [shape1, shape2],
+        }
+    )
+    i = 0
+    with meas_with_registered_param_2d.run(
+        dataset_class=DataSetType.DataSetInMem
+    ) as datasaver:
+        for set_v in np.linspace(0, 25, shape1):
+            for set_v2 in np.linspace(0, 100, shape2):
+                DAC.ch1.set(set_v)
+                DAC.ch2.set(set_v2)
+                datasaver.add_result(
+                    (DAC.ch1, set_v), (DAC.ch2, set_v2), (DMM.v1, float(i))
+                )
+                i = i + 1
+    ds = datasaver.dataset
+    ds.add_metadata("mymetadatatag", 42)
+
+    paramspecs = ds.get_parameters()
+    assert len(paramspecs) == 3
+    assert paramspecs[0].name == "dummy_dac_ch1"
+    assert paramspecs[1].name == "dummy_dac_ch2"
+    assert paramspecs[2].name == "dummy_dmm_v1"
+
+    # if the indexes are not correct here it will break the exported
+    # xarray and therefor netcdf order below and therefor the loaded data
+    # will have the coordinates inverted.
```

Comment:
```suggestion
    # if the indexes (their order) are not correct here, the exported xarray, and thus
    # the exported netcdf will have a wrong order of axes in the data, so that
    # the loaded data will have the coordinates inverted. Hence we assert that
    # the order is exactly the same as declared via Measurement.register_parameter
    # calls above
```